src/test/java/fi/otavanopisto/kuntaapi/test/server/integration/gtfs/TripTestsIT.java

```diff
@@ -0,0 +1,138 @@
+  
+  @Before
+  public void beforeTest() throws InterruptedException {
+    createPtvSettings();
+    
+    getPtvMocker()
+      .mockOrganizations("0de268cf-1ea1-4719-8a6e-1150933b6b9e")
+      .startMock();
+
+    waitApiListCount("/organizations", 1);
+    
+    createGtfsSettings(getOrganizationId(0));
+
+    waitApiListCount(String.format("/organizations/%s/transportTrips", getOrganizationId(0)), 7); 
+  }
+
+  @After
+  public void afterClass() {
+    String organizationId = getOrganizationId(0);
+    getPtvMocker().endMock();
+    deletePtvSettings();
+    deleteGtfsSettings(organizationId);
+  }
+
+  @Test
+  public void testListTrips() {
+    String organizationId = getOrganizationId(0);
+    
+    Response response = given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/transportTrips", organizationId);
+    
+    response
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(7))
+      .body("id[1]", notNullValue())
+      .body("routeId[1]", notNullValue())
+      .body("scheduleId[1]", notNullValue())
+      .body("headsign[1]", is("Mikkeli matkakeskus"));
+    
+    String routeId = response.body().jsonPath().getString("routeId[1]");
+    String ScheduleId = response.body().jsonPath().getString("scheduleId[1]");
+    
+    assertFound(String.format("/organizations/%s/transportRoutes/%s", organizationId, routeId));
+    assertFound(String.format("/organizations/%s/transportSchedules/%s", organizationId, ScheduleId));
+    
+  }
+  
+  @Test
+  public void testFindTrip() {
+    String organizationId = getOrganizationId(0);
+    Response response = given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/transportTrips/{tripId}", organizationId, getOrganizationTripId(organizationId, 2));
+    
+    response
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id", notNullValue())
+      .body("routeId", notNullValue())
+      .body("scheduleId", notNullValue())
+      .body("headsign", is("Pajulankylä"));
+    
+    String routeId = response.body().jsonPath().getString("routeId");
+    String ScheduleId = response.body().jsonPath().getString("scheduleId");
+    
+    assertFound(String.format("/organizations/%s/transportRoutes/%s", organizationId, routeId));
+    assertFound(String.format("/organizations/%s/transportSchedules/%s", organizationId, ScheduleId));
+  }
+  
+  @Test
+  public void testOrganizationTripsNotFound() throws InterruptedException {
+    String organizationId = getOrganizationId(0);
+    String incorrectOrganizationId = getOrganizationId(1);
+    String organizationTripId = getOrganizationTripId(organizationId, 0);
+    
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    assertFound(String.format("/organizations/%s/transportTrips/%s", organizationId, organizationTripId));
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "/organizations/%s/transportTrips/%s" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)